tests/components/plex/test_config_flow.py

```diff
@@ -0,0 +1,265 @@
+    """Test importing a legacy file during discovery."""
+
+    mock_file_contents = {
+        "1.2.3.4:32400": {"ssl": False, "token": "12345", "verify": True}
+    }
+    file_host_and_port, file_config = list(mock_file_contents.items())[0]
+    used_url = f"http://{file_host_and_port}"
+
+    with patch("plexapi.server.PlexServer") as mock_plex_server, patch(
+        "homeassistant.components.plex.config_flow.load_json",
+        return_value=mock_coro(mock_file_contents),
+    ):
+        type(mock_plex_server.return_value).machineIdentifier = PropertyMock(
+            return_value="unique_id_123"
+        )
+        type(mock_plex_server.return_value).friendlyName = PropertyMock(
+            return_value="Mock Server"
+        )
+        type(mock_plex_server.return_value)._baseurl = PropertyMock(
+            return_value=used_url
+        )
+
+        result = await hass.config_entries.flow.async_init(
+            config_flow.DOMAIN,
+            context={"source": "discovery"},
+            data={CONF_HOST: "1.2.3.4", CONF_PORT: "32400"},
+        )
+        print(result)
+
+        assert result["type"] == "create_entry"
+        assert result["title"] == "Mock Server"
+        assert result["data"][config_flow.CONF_SERVER] == "Mock Server"
+        assert result["data"][config_flow.CONF_SERVER_IDENTIFIER] == "unique_id_123"
+        assert result["data"][config_flow.PLEX_SERVER_CONFIG][CONF_URL] == used_url
+        assert (
+            result["data"][config_flow.PLEX_SERVER_CONFIG][CONF_TOKEN]
+            == file_config[CONF_TOKEN]
+        )
+
+
+async def test_discovery(hass, aioclient_mock):
+    """Test starting a flow from discovery."""
+
+    result = await hass.config_entries.flow.async_init(
+        config_flow.DOMAIN,
+        context={"source": "discovery"},
+        data={CONF_HOST: "1.2.3.4", CONF_PORT: "32400"},
+    )
+
+    assert result["type"] == "form"
+    assert result["step_id"] == "user"
+
+
+async def test_import_bad_hostname(hass, aioclient_mock):
+    """Test when an invalid address is provided."""
+
+    with patch(
+        "plexapi.server.PlexServer", side_effect=requests.exceptions.ConnectionError
+    ):
+        result = await hass.config_entries.flow.async_init(
+            config_flow.DOMAIN,
+            context={"source": "import"},
+            data={CONF_TOKEN: "12345", CONF_URL: "http://1.2.3.4:32400"},
+        )
+
+        assert result["type"] == "form"
+        assert result["step_id"] == "user"
+        assert result["errors"]["base"] == "not_found"
+
+
+async def test_unknown_exception(hass, aioclient_mock):
+    """Test when an unknown exception is encountered."""
+
+    result = await hass.config_entries.flow.async_init(
+        config_flow.DOMAIN, context={"source": "user"}
+    )
+
+    assert result["type"] == "form"
+    assert result["step_id"] == "user"
+
+    with patch("plexapi.myplex.MyPlexAccount", side_effect=Exception):
```

Comment:
Is this the exception that is raised?

Reply:
There's a generic `Exception` fallback catch in `config_flow`. Maybe that shouldn't be there...

Reply:
It's ok to have a generic fallback if we also catch other specific exceptions.